pylint/checkers/typecheck.py

```diff
@@ -545,7 +546,7 @@ def _emit_no_member(node, owner, owner_name, ignored_mixins=True, ignored_none=T
                 )
             ):
                 return False
-        node_origin, parent = parent, parent.parent
```

Comment:
@Pierre-Sassoulas Is this the one you're talking about? I agree that it's a little complicated, but I would say that was _already_ the case, especially with this double-assignment parent-child walk-up maneuver here. Getting rid of that IMO justifies adding in the break logic. But I can take it out if you feel strongly about it.

Maybe a way to make it clearer would be to return false straightaway instead of breaking the loop.

Reply:
I don't have strong opinion, let's see what Marc think.

Reply:
It works either way, with `while` and with `for`. However I think the previous code was a bit easier to understand in this case. I especially like the tuple assignment as it makes it clear what actually happens.

If I remember correctly, I used something similar in another helper method.

--
A change to an early `return` wouldn't change much. Instead it would only create the possibility to add new errors.

Reply:
--
Obviously, this would be a good spot for the new helper.